bundle/edu.gemini.sp.vcs/src/main/scala/edu/gemini/sp/vcs2/MergePlan.scala

```diff
@@ -35,10 +35,14 @@ case class MergePlan(update: Tree[MergeNode], delete: Set[Missing]) {
 
   /** Gets the `VersionMap` of the provided program as it will be after the
     * updates in this plan have been applied. */
-  def vm(p: ISPProgram): VersionMap = {
+  def vm(p: ISPProgram): VersionMap = vm(p.getVersions)
+
+  /** Gets the `VersionMap` with any modifications required that will be made
+    * by this merge plan. */
+  def vm(vm0: VersionMap): VersionMap = {
     // Extract the updates to the VersionMap from the MergePlan.
     val vmUpdates: VersionMap = {
-      val vm0 = update.foldRight(Map.empty[SPNodeKey, NodeVersions]) { (mn, m) =>
+      val vm0 = update.sFoldRight(Map.empty[SPNodeKey, NodeVersions]) { (mn, m) =>
```

Comment:
I'm glad you called this out.  I've had a couple of problems with folding a `scalaz.Tree`.  At first I worked around this by providing a version of `foldRight` that evaluates its result eagerly using this signature:
````scala
  def sFoldRight[B](z: B)(f: (A, B) => B): B
````
and simply calling the provided `foldRight`:
````scala
def foldRight[B](z: => B)(f: (A, => B) => B): B 
````
For giant programs though this was still causing a stack overflow so I switched to my own simplistic `@tailrec` implementation and the problems went away.  Still it is not very satisfying that the library-provided `foldRight` chokes on large inputs.